crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -1842,6 +1840,28 @@ impl<'a> Checker<'a> {
         self.semantic.flags = snapshot;
     }
 
+    /// Visit an [`Expr`], and treat it as a [PEP 613] explicit type alias.
+    ///
+    /// [PEP 613]: https://peps.python.org/pep-0613/
+    fn visit_explicit_type_alias(&mut self, expr: &'a Expr) {
+        let snapshot = self.semantic.flags;
+        self.semantic.flags |= SemanticModelFlags::EXPLICIT_TYPE_ALIAS;
+        self.visit_type_definition(expr);
+        self.semantic.flags = snapshot;
+    }
+
+    /// Visit an [`Expr`], and treat it as a [PEP 695] generic type alias.
+    ///
+    /// [PEP 695]: https://peps.python.org/pep-0695/#generic-type-alias
+    fn visit_generic_type_alias(&mut self, expr: &'a Expr) {
+        let snapshot = self.semantic.flags;
+        self.semantic.flags |= SemanticModelFlags::GENERIC_TYPE_ALIAS;
+        self.visit
+            .type_param_definitions
+            .push((expr, self.semantic.snapshot()));
+        self.semantic.flags = snapshot;
```

Comment:
I'm sorry if this is a dumb question, but I'm not very familiar with this part of the code base.

I suspect that mutating `self.semantic.flags` is necessary so that the flags are part of the snapshot created by `self.semantic.snapshot`? This wasn't immediately clear to me. It might be worth adding a comment explaining this.